add Succession schema for task_params to environment

wizardofoz123 opened the pull request:
This partially implements FASTMOVING-833.

Let's not depend on Beaker terminology, that makes it hard to move
forward.

Signed-off-by: Jakub Racek <jracek@redhat.com>

spbnick commented on kpet/data.py:
```diff
@@ -16,6 +16,8 @@
 from kpet.schema import Invalid, Struct, Reduction, NonEmptyList, \
     List, Dict, String, Regex, ScopedYAMLFile, YAMLFile, Class, Boolean, Int
+from kpet.schema import Succession
+
```
Can you just add this to the list directly above?

wizardofoz123 replied:
I could, but doesn't that make diffs more confusing? When each imported thing has its own line, it's clearer to see what was added or removed.

spbnick replied:
Yeah, makes sense, go ahead. I just don't want a ton of lines, for this, but if it's easier for you, sure. I just do `git diff --color-words=.` for intra-line differences.

spbnick commented on kpet/data.py:
```diff
@@ -203,34 +205,66 @@ def __init__(self, data):
+                        role=String(),
+                        url_suffix=String(),
+                        environment=Dict(String()),
+                    )
```
Looking at this, it might be better to just add support for aliases to `Struct`. E.g. have it accept the optional `aliases` argument, and let that be a string-string (alias-canonical name) dictionary, where keys must not be present in either `required` or `optional` and values must be. Then update `validate` to recognize aliases as the canonical fields, making sure both are not present, and update `resolve` to rename aliases to canonical fields.

What do you think?

Although, it is of course faster to do this instead, at this moment, we might save time on future renames.

Otherwise this looks OK. Only maybe put down a TODO to remove the succession once the database is updated.

spbnick replied:
Another (probably simplest) way could be just to add `environment` as an optional field, and do renaming in-class. Then remove the `task_params` field once kpet-db has transitioned. This way you won't have to copy the whole schema either.

wizardofoz123 replied:
I was planning to remove the whole Succession schema usage from here once kpet-db changes are done. Is that sufficient? I'd like to take this approach.

spbnick replied:
Sure, I understood what you were going to do. Just thought copy-pasting large structures like this gets tedious and error-prone fast and we might have a better approach. But it's OK as is for now.

spbnick reviewed (commented):
Thanks for taking this on! Looks good, just have a couple questions/suggestions.

Commit:
add Succession schema for task_params to environment

This partially implements FASTMOVING-833.

Let's not depend on Beaker terminology, that makes it hard to move
forward.

Signed-off-by: Jakub Racek <jracek@redhat.com>
